test/model/message_list_test.dart

```diff
@@ -271,6 +271,42 @@ void main() {
     checkInvariants(model);
   });
 
+  group('DeleteMessageEvent', () {
+    final stream = eg.stream();
+    final messages = List.generate(30, (i) => eg.streamMessage(stream: stream));
+
+    test('in narrow', () async {
```

Comment:
Let's also have a test case where the deleted messages are non-consecutive, and don't include the first or last message. That's one of the things that makes the logic complicated.

Also where all the messages get deleted. That seems like the sort of thing that a buggy implementation could get something wrong on.